web/src/providers/PasskeyProvider.tsx

```diff
@@ -0,0 +1,235 @@
+'use client';
+
+import React, { createContext, useContext, useState, useEffect, useCallback, useMemo } from 'react';
+import {
+  createKernelAccount,
+  createKernelAccountClient,
+  createZeroDevPaymasterClient,
+  KernelAccountClient,
+  KernelSmartAccount,
+  SponsorUserOperationParameters,
+} from '@zerodev/sdk';
+import {
+  toPasskeyValidator,
+  toWebAuthnKey,
+  WebAuthnMode,
+  PasskeyValidatorContractVersion,
+  deserializePasskeyValidator,
+} from '@zerodev/passkey-validator';
+import { KERNEL_V3_1 } from '@zerodev/sdk/constants';
+import { ENTRYPOINT_ADDRESS_V07 } from 'permissionless';
+import { http, createPublicClient, Hex, Transport, Chain, Address } from 'viem';
+import { getPasskeyData, setPasskeyData } from '@/utils/passkey';
+import { getChainsForEnvironment } from '@/store/supportedChains';
+import storage from 'local-storage-fallback';
+
+const chain = getChainsForEnvironment();
+const publicClient = createPublicClient({
+  chain: chain,
+  transport: http(),
+});
+
+const entryPoint = ENTRYPOINT_ADDRESS_V07;
+const kernelVersion = KERNEL_V3_1;
+
+const zerodevApiKey = process.env.NEXT_PUBLIC_ZERODEV_API_KEY;
+const passkeyServer = `https://passkeys.zerodev.app/api/v4`;
+const paymasterUrl = `https://rpc.zerodev.app/api/v2/paymaster/${zerodevApiKey}`;
+const bundlerRpc = `https://rpc.zerodev.app/api/v2/bundler/${zerodevApiKey}`;
+
+type PasskeyContextType = {
+  address: Hex | undefined;
+  isConnecting: boolean;
+  isInitializing: boolean;
+  account: KernelSmartAccount<typeof ENTRYPOINT_ADDRESS_V07, Transport, Chain> | null;
+  accountClient: KernelAccountClient<typeof ENTRYPOINT_ADDRESS_V07, Transport, Chain> | undefined;
+  login: () => Promise<void>;
+  register: () => Promise<void>;
+  logout: () => void;
+};
+
+// Create a default context value
+const defaultContextValue: PasskeyContextType = {
+  address: undefined,
+  isConnecting: false,
+  isInitializing: true,
+  account: null,
+  accountClient: undefined,
+  login: async () => {},
+  register: async () => {},
+  logout: () => {},
+};
+
+// Use the default context value when creating the context
+const PasskeyContext = createContext<PasskeyContextType>(defaultContextValue);
+
+export function PasskeyProvider({ children }: { children: React.ReactNode }) {
+  const [address, setAddress] = useState<Hex | undefined>(undefined);
```

Comment:
Can consider to have a PasskeyState to manage all the related state here.

Reply:
Claude said this:

Given your current setup and the relatively small number of state variables, I believe the current approach with separate states is more appropriate and maintainable. It provides a good balance between clarity, flexibility, and ease of use. If your application grows significantly in complexity, you can always refactor to a single state object or even consider using a more robust state management solution like Redux or MobX.